gef.py

```diff
@@ -1605,15 +1605,15 @@ class RISCV(Architecture):
     arch = "RISCV"
     mode = "RISCV"
 
-    all_registers = ["$zero", "$ra", "$sp", "$gp", "$x4", "$t0", "$t1",
-                     "$t2", "$fp", "$s1", "$a1", "$a2", "$a3", "$a4",
-                     "$a5", "$a6", "$a7", "$s2", "$s3", "$s4", "$s5",
-                     "$s6", "$s7", "$s8", "$s9", "$s10", "$s11", "$t3",
-                     "$t4", "$t5", "$t6",]
+    all_registers = ["$zero", "$ra", "$sp", "$gp", "$tp", "$t0", "$t1",
+                     "$t2", "$fp", "$s1", "$a0", "$a1", "$a2", "$a3",
+                     "$a4", "$a5", "$a6", "$a7", "$s2", "$s3", "$s4",
+                     "$s5", "$s6", "$s7", "$s8", "$s9", "$s10", "$s11",
+                     "$t3", "$t4", "$t5", "$t6",]
     return_register = "$a0"
     function_parameters = ["$a0", "$a1", "$a2", "$a3", "$a4", "$a5", "$a6", "$a7"]
     syscall_register = "$a7"
-    syscall_register = "ecall"
+    syscall_instructions = ["ecall"]
```

Comment:
Oops how did we let that through.

Reply:
Not sure... my bad :man_facepalming:

Reply:
@Grazfather we don't have automated tests for RISC-V, that could be a beginning of the "how"...